### ISSUE TITLE
Registering or Unregistering provider Microsoft.AVS

### ISSUE THREAD

**Issue by soockee** [Azure/azure-cli#28349]
### Describe the bug

Trying to register "Microsoft.AVS" for my subscription via the portal but it gets stuck in registering and eventually gives this kind of message "Unregister' operation check timed out on Resource Provider 'microsoft.AVS', please refresh resource providers list to check for registration status".

### Related command

az provider register --namespace  Microsoft.AVS --verbose --wait

### Errors

no error, but very slow (hours? days?)

### Issue script & Debug output

```
az provider list --query "[?registrationState=='Unregistering']" -o table                                                                                         3s ☁ 
Namespace      RegistrationState    RegistrationPolicy
-------------  -------------------  --------------------
Microsoft.AVS  Unregistering        RegistrationRequired
```

### Expected behavior

Provider should be registered in the subscription

### Environment Summary

{
  "azure-cli": "2.57.0",
  "azure-cli-core": "2.57.0",
  "azure-cli-telemetry": "1.1.0",
  "extensions": {
    "azure-devops": "0.26.0"
  }
}

### Additional context

Identical Symptoms to [28203](https://github.com/Azure/azure-cli/issues/28203), just different provider

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by jodok** [comment]
same here. stuck in registering state since 1-2 days

**Comment by spaelling** [comment]
same here but with Terraform. Not that it should make any difference.

Tried both with EA subscription and a PAYG. Registration never seems to finish.

PAYG done in portal which results in an error

```
'Register' operation check timed out on Resource Provider 'microsoft.avs', please refresh resource providers list to check for registration status
```

status is still registering.

**Comment by soockee** [comment]
@spaelling 

> same here but with Terraform. Not that it should make any difference.

A _possible_ workaround  if your code is not dependent on the functionality of this azure provider and you are using the `azurerm` terraform provider. This could provide value if you are working with playground envs or personal stuff:

```
provider "azurerm" {
  #  Azure infinitly tries to register a azure resource provider "Microsoft.AVS" 
  # Therefore skip registration 
  skip_provider_registration = true 
  features {}
  }
}
```

**Comment by spaelling** [comment]
> @spaelling
> 
> > same here but with Terraform. Not that it should make any difference.
> 
> A _possible_ workaround if your code is not dependent on the functionality of this azure provider and you are using the `azurerm` terraform provider. This could provide value if you are working with playground envs or personal stuff:
> 
> ```
> provider "azurerm" {
>   #  Azure infinitly tries to register a azure resource provider "Microsoft.AVS" 
>   # Therefore skip registration 
>   skip_provider_registration = true 
>   features {}
>   }
> }
> ```

yes, i did that already. No clue why it registers that provider without needing it. I actually had to look up what it was :D

**Comment by mortenjoenby** [comment]
@spaelling , so what is this provider? Something with VMware on Azure - right?
I find it difficult to find good documentation on the providers.

**Comment by spaelling** [comment]
> @spaelling , so what is this provider? Something with VMware on Azure - right?
> 
> I find it difficult to find good documentation on the providers.

Yes, it's the VMWare provider. And  the documentation on resource providers could be better or easier to find.

**Comment by piankris** [comment]
I have the same issue. The status is still at "Registering" even after 3 days...

**Comment by gabrimonfa** [comment]
Same here, cannot register, cannot unregister. This is annoying to say the least

**Comment by mopopoola** [comment]
I am having the same issue. I'm trying to register Microsoft.AVS resource provider and it gets stuck on registering. Tried many times since Feb 10 and today is Feb 12. Getting error message 'register' operation check timed out on Resource provider 'microsoft.avs'. please refresh resource providers list to check for registration status.

**Comment by jasonkind** [comment]
I'm having the same issue trying to register Microsoft.Storage.  It was stuck in the Registering state so I tried to unregister it.  Now it's stuck in the Unregistering state.  Has anyone figured out a resolution?

**Comment by piankris** [comment]
> I have the same issue. The status is still at "Registering" even after 3 days...

It now shows as registered for me but a week later. The provider `Microsoft.ResourceNotifications` that I tried to register the same day as `Microsoft.AVS` is still pending (Registering)... It's been a week now.
